File: src/main/java/com/ada/conversation/context/ToolResponseContextItem.java
package com.ada.conversation.context;

import com.ada.conversation.application.dto.*; import com.ada.shared.observability.MeasuredContextItem; import org.springframework.core.annotation.Order; import org.springframework.stereotype.Component;
@Component @Order(50) @MeasuredContextItem("tool_response") public class ToolResponseContextItem implements ContextItem { public LlmContentComponent component(){return LlmContentComponent.TOOL_RESPONSE;} public ContextState apply(ChatRequest r,ContextState c){return c;} }

File: src/main/kotlin/com/ada/conversation/context/ToolResponseContextItem.kt
package com.ada.conversation.context

import com.ada.conversation.application.dto.ChatRequest
import com.ada.conversation.application.dto.ContextState
import com.ada.conversation.application.dto.LlmContentComponent
import org.springframework.core.annotation.Order
import org.springframework.stereotype.Component
import com.ada.shared.observability.MeasuredContextItem

@Component
@Order(50)
@MeasuredContextItem("tool_response")
class ToolResponseContextItem : ContextItem {
    override val component = LlmContentComponent.TOOL_RESPONSE

    override fun apply(request: ChatRequest, current: ContextState): ContextState = current
}
